client-s3/src/main/kotlin/app/cash/backfila/client/s3/record/StringStrategies.kt

```diff
@@ -0,0 +1,24 @@
+package app.cash.backfila.client.s3.record
+
+import okio.BufferedSource
+import okio.ByteString
+
+// This file contains some general strategies for handling String files for Backfills.
+
+/**
+ * Produces single string records separated by newlines.
+ */
+class Utf8StringNewlineStrategy : RecordStrategy<String> {
+  override fun calculateNextRecordBytes(source: BufferedSource): Long = newlineByteCalculator(source)
```

Comment:
what happens if you have a very long line in a file, I'm assuming the buffer here is not a full file

Reply:
I think maybe you'd want a different strategy for splitting up the file in that case?

Reply:
then what's the line length limit supported, can we document it here?

Reply:
I'm pretty confident we don't actually want to limit this. The one thing we do check is that your strategy does always consume part of the stream so we are making progress.